website/content/docs/secrets/terraform.mdx

```diff
@@ -0,0 +1,167 @@
+    Terraform Cloud [Account
+    API](https://www.terraform.io/docs/cloud/api/account.html) to find the
+    desired User ID.
+
+    ```text
+    $ vault write terraform/roles/my-role user_id=user-12345abcde
+    Success! Data written to: terraform/roles/my-role
+    ```
+
+## Usage
+
+After the secrets engine is configured and a user/machine has a Vault token with
+the proper permission, it can generate credentials.
+
+Generate a new credential by reading from the `/creds` endpoint with the name
+of the role:
+
+```shell-session
+$ vault read terraform/creds/my-role
+Key                Value
+---                -----
+lease_id           terraform/creds/my-user/A_LEASE_ID_PdvmJjACTtKrY2I
+lease_duration     180s
+lease_renewable    true
+token              TJFDSIFDSKFEKZX.FKFKA.akjlfdiouajlkdakadfiowe
+token_id           at-123acbdfask
+```
+
+## Organization, Team, and User Roles
+
+Terraform Cloud supports three distinct types of API tokens; Organizations,
+Teams, and Users. Each token type has distinct access levels and generation
+workflows. A given Vault role can manage any one of the three types at a time,
+however there are important differences to be aware of.
+
+### Organization and Team roles 
+
+The Terraform Cloud API limits both Organization and Team roles to **one active
+token at any given time**. Generating a new Organization or Team API token by
+reading the credentials in Vault or otherwise generating them on
+[app.terraform.io](https://app.terraform.io) will effectively revoke **any**
+existing API token for that Organization or Team. 
+
+Due to this behavior, Organization and Team API tokens created by Vault will be
+stored and returned on future requests, until the credentials get rotated. This
+is to prevent unintentional revocation of tokens that are currently in-use.
+
+Below is an example of creating a Vault role to manage manage an Organization
+API token and rotating the token:
+
+```shell-session
+$ vault write terraform/role/testing organization="${TF_ORGANIZATION}"
+Success! Data written to: terraform/role/testing
+
+$ vault write -f terraform/rotate-role/testing
+Success! Data written to: terraform/rotate-role/testing
+```
+
+The API token is retrieved by reading the credentials for the role:
+
+```
+$ vault read terraform/creds/testing
+
+Key             Value
+---             -----
+organization    hashicorp-vault-testing
+role            testing
+token           <example token>
+token_id        at-fqvtdTQ5kQWcjUfG
+```
+
+### User roles 
+
+Traditionally Vault secret engines creates dynamic users and dynamic credentials
+along with them. At the time of writing, the Terraform Cloud API does not allow
+for creating dynamic users. Instead, the Terraform Cloud secret engine creates
+dynamice User API tokens by configuring a Vault role to manage an existing
+Terraform Cloud user. The lifecycle of these tokens is managed by Vault and
+will auto expire according to the configured TTL and Max TTL of the Vault
```

Comment:
Should `Max` be capitalized? Feel free to ignore this one.

Reply:
Probably not, good catch. Fixed in e6135a9c7